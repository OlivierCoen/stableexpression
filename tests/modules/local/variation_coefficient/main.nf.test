nextflow_process {

    name "Test Process VARIATION_COEFFICIENT"
    script "modules/local/variation_coefficient/main.nf"
    process "VARIATION_COEFFICIENT"
    tag "var_coeff"
    tag "module"

    test("Should run without failures") {

        when {

            process {
                """
                ch_renamed_files = Channel.fromPath( '$baseDir/tests/input/variation_coefficient/*_renamed.csv', checkIfExists: true)
                ch_metadata_files = Channel.fromPath( '$baseDir/tests/input/variation_coefficient/*_metadata.csv', checkIfExists: true)
                ch_mapping_files = Channel.fromPath( '$baseDir/tests/input/variation_coefficient/*_mapping.csv', checkIfExists: true)
                input[0] = ch_renamed_files.collect()
                input[1] = ch_metadata_files.collect()
                input[2] = ch_mapping_files.collect()
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
