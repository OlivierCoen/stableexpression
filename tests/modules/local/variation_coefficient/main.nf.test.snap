{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,329ed4c3197a7b48c8f5ef47f35caa1e"
                ],
                "1": [
                    [
                        "VARIATION_COEFFICIENT",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "csv": [
                    "variation_coefficients.csv:md5,329ed4c3197a7b48c8f5ef47f35caa1e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-01T20:40:24.443926287"
    }
}